pcsx2/USB/usb-pad/usb-train.cpp

```diff
@@ -0,0 +1,555 @@
+		switch (s->type)
+		{
+			case TRAIN_TYPE2:
+			{
+				TrainConData_Type2 out = {};
+				out.control = 0x1;
+				out.brake = s->passthrough ? s->data.brake : dct01_brake(s->data.brake);
+				out.power = s->passthrough ? s->data.power : dct01_power(s->data.power);
+				out.horn = 0xFF; // Button C doubles as horn.
+				out.hat = s->data.hatswitch;
+				out.buttons = dct01_buttons(s->data.buttons);
+				usb_packet_copy(p, &out, sizeof(out));
+				break;
+			}
+			case TRAIN_SHINKANSEN:
+			{
+				TrainConData_Shinkansen out = {};
+				out.brake = s->passthrough ? s->data.brake : dct02_brake(s->data.brake);
+				out.power = s->passthrough ? s->data.power : dct02_power(s->data.power);
+				out.horn = 0xFF; // Button C doubles as horn, skip.
+				out.hat = s->data.hatswitch;
+				out.buttons = dct02_buttons(s->data.buttons);
+				usb_packet_copy(p, &out, sizeof(out));
+				break;
+			}
+			case TRAIN_RYOJOUHEN:
+			{
+				TrainConData_Ryojouhen out = {};
+				out.brake = s->passthrough ? s->data.brake : dct03_brake(s->data.brake);
+				out.power = s->passthrough ? s->data.power : dct03_power(s->data.power);
+				out.horn = 0xFF; // Dedicated horn button, skip.
+				out.hat = s->data.hatswitch & 0x0F;
+				out.buttons = dct03_buttons(s->data.buttons);
+				usb_packet_copy(p, &out, sizeof(out));
+				break;
+			}
+			default:
+				Console.Error("Unhandled TrainController USB_TOKEN_IN pid=%d ep=%u type=%u", p->pid, p->ep->nr, s->type);
+				p->status = USB_RET_IOERROR;
+				return;
+		}
+	}
+
+	USBDevice* TrainDevice::CreateDevice(SettingsInterface& si, u32 port, u32 subtype) const
+	{
+		TrainDeviceState* s = new TrainDeviceState(port, static_cast<TrainDeviceTypes>(subtype));
+
+		s->desc.full = &s->desc_dev;
+
+		switch (subtype)
+		{
+			case TRAIN_TYPE2:
+				s->desc.str = dct01_desc_strings;
+				if (usb_desc_parse_dev(dct01_dev_descriptor, sizeof(dct01_dev_descriptor), s->desc, s->desc_dev) < 0)
+					goto fail;
+				break;
+			case TRAIN_SHINKANSEN:
+				s->desc.str = dct02_desc_strings;
+				if (usb_desc_parse_dev(dct02_dev_descriptor, sizeof(dct02_dev_descriptor), s->desc, s->desc_dev) < 0)
+					goto fail;
+				break;
+			case TRAIN_RYOJOUHEN:
+				s->desc.str = dct03_desc_strings;
+				if (usb_desc_parse_dev(dct03_dev_descriptor, sizeof(dct03_dev_descriptor), s->desc, s->desc_dev) < 0)
+					goto fail;
+				break;
+
+			default:
+				goto fail;
+		}
+
+		if (usb_desc_parse_config(taito_denshacon_config_descriptor, sizeof(taito_denshacon_config_descriptor), s->desc_dev) < 0)
+			goto fail;
+
+		s->dev.speed = USB_SPEED_FULL;
+		s->dev.klass.handle_attach = usb_desc_attach;
+		s->dev.klass.handle_reset = train_handle_reset;
+		s->dev.klass.handle_control = train_handle_control;
+		s->dev.klass.handle_data = train_handle_data;
+		s->dev.klass.unrealize = train_handle_destroy;
+		s->dev.klass.usb_desc = &s->desc;
+		s->dev.klass.product_desc = s->desc.str[2];
+
+		usb_desc_init(&s->dev);
+		usb_ep_init(&s->dev);
+		train_handle_reset(&s->dev);
```

Comment:
The passthrough value isn't read in CreateDevice like this : 
https://github.com/PCSX2/pcsx2/blob/fd5a6522702c3279c2a968f395983a46374b7fa2/pcsx2/USB/usb-lightgun/guncon2.cpp#L469
This is why passthrough axes are messed up.

Reply:
This should fix DGOC-44U

Reply:
Oh, that was an oversight on my part. Nice find, will fix this up.